L1Trigger/Phase2L1Taus/plugins/L1HPSPFTauProducer.cc

```diff
@@ -0,0 +1,278 @@
+#include "L1Trigger/Phase2L1Taus/interface/L1HPSPFTauProducer.h"
+
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include <cmath> // std::fabs
+
+L1HPSPFTauProducer::L1HPSPFTauProducer(const edm::ParameterSet& cfg) 
+  : moduleLabel_(cfg.getParameter<std::string>("@module_label"))
+  , tauBuilder_(nullptr)
+  , useChargedPFCandSeeds_(cfg.getParameter<bool>("useChargedPFCandSeeds"))
+  , min_seedChargedPFCand_pt_(cfg.getParameter<double>("min_seedChargedPFCand_pt"))
+  , max_seedChargedPFCand_eta_(cfg.getParameter<double>("max_seedChargedPFCand_eta"))
+  , max_seedChargedPFCand_dz_(cfg.getParameter<double>("max_seedChargedPFCand_dz"))
+  , usePFJetSeeds_(cfg.getParameter<bool>("usePFJetSeeds"))
+  , min_seedPFJet_pt_(cfg.getParameter<double>("min_seedPFJet_pt"))
+  , max_seedPFJet_eta_(cfg.getParameter<double>("max_seedPFJet_eta"))
+  , min_PFTau_pt_(cfg.getParameter<double>("min_PFTau_pt"))
+  , max_PFTau_eta_(cfg.getParameter<double>("max_PFTau_eta"))
+  , min_leadChargedPFCand_pt_(cfg.getParameter<double>("min_leadChargedPFCand_pt"))
+  , max_leadChargedPFCand_eta_(cfg.getParameter<double>("max_leadChargedPFCand_eta"))
+  , max_leadChargedPFCand_dz_(cfg.getParameter<double>("max_leadChargedPFCand_dz"))
+  , max_chargedIso_(cfg.getParameter<double>("max_chargedIso"))
+  , max_chargedRelIso_(cfg.getParameter<double>("max_chargedRelIso"))
+  , deltaR_cleaning_(cfg.getParameter<double>("deltaR_cleaning"))
+  , applyPreselection_(cfg.getParameter<bool>("applyPreselection"))
+  , debug_(cfg.getUntrackedParameter<bool>("debug", false))
+{
+  if ( debug_ )
+  {
+    std::cout << "<L1HPSPFTauProducer::L1HPSPFTauProducer (moduleLabel = " << moduleLabel_ << ")>:" << std::endl;
+  }
+
+  tauBuilder_ = new L1HPSPFTauBuilder(cfg);
```

Comment:
@sandeepbhowmik1 
Please make a follow-up PR with replacing this with a smart pointer.